feat: automatic check for updates

szwabodev opened the pull request:
- added toggle in Settings->Server->Updates dialog for automatic server update checking (enabled by default, saved in local storage)
- use interval (15min) to periodically check for new updates (if enabled via toggle)
- check for new updates on page load (if enabled via toggle)
- show "Update available" button in header when new server update is available
- improved logic for checking available updates to use Docker hub API instead of pulling image, which makes checking update a lot faster
- moved pulling latest release to `updateServer` action
- removed green dot pinging while server is updating
- added button loading text while server is updating
- added automatic page reload on successful server update
- added information about page reload in confirmation modal and success toast
- added `getDokployImageTag` helper
- improved toasts messages related with updating to be clearer (separate for error, available, unavailable)

Screenshots:
![image](https://github.com/user-attachments/assets/8d6fff95-49da-4d4f-b863-df5a2a1a9041)
![image](https://github.com/user-attachments/assets/1e0056e2-f49c-4b2d-a025-9250566f933f)
![image](https://github.com/user-attachments/assets/da57de34-ed02-4ee5-830d-59cdd4f27128)
![image](https://github.com/user-attachments/assets/3c178697-2a30-4c00-a190-ea6b08039001)
![image](https://github.com/user-attachments/assets/7837b7a6-a2e4-4a8b-90ca-5d61508333d1)

drudge commented:
It would be great if this modal showed what the version was.

Siumauricio commented:
@szwabodev Hi, I'm getting this error 

![Screenshot 2024-12-20 at 11 33 01 PM](https://github.com/user-attachments/assets/719ed03f-2538-4dff-8d04-3caaf14a0b45)
![Screenshot 2024-12-20 at 11 33 41 PM](https://github.com/user-attachments/assets/e766da6c-00ac-4af0-a2a2-8961f86781e2)

szwabodev commented:
> @szwabodev Hi, I'm getting this error
> 
> ![Screenshot 2024-12-20 at 11 33 01 PM](https://private-user-images.githubusercontent.com/47042324/397906972-719ed03f-2538-4dff-8d04-3caaf14a0b45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzAzMDMsIm5iZiI6MTczNDc3MDAwMywicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDY5NzItNzE5ZWQwM2YtMjUzOC00ZGZmLThkMDQtM2NhYWYxNGEwYjQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA4MzMyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxZDJjZTY3ODY5MWRkYTJjYjQwNDBkM2JjOGRlZDY1ZmM3YTEyOGZhYzJhMWM1Y2I4OTcyMWEzMWJmNmU3NTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cyQwZnmFRrw9XFp3Hwlz3eHHRbUalfwYjVB-968Mty4) ![Screenshot 2024-12-20 at 11 33 41 PM](https://private-user-images.githubusercontent.com/47042324/397907000-e766da6c-00ac-4af0-a2a2-8961f86781e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzAzMDMsIm5iZiI6MTczNDc3MDAwMywicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDcwMDAtZTc2NmRhNmMtMDBhYy00YWYwLWEyYTItODk2MWY4Njc4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA4MzMyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhY2I0ZTRlOWM2MGU5YTJmMTJjN2ZiNTI0MTUxNDExODlhOTJiMmQ1YTA0NDI3MDJhZjM0ZGEzNmQ5ZDllODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ctrb7wmPbaP77a0SNc_pXct_Lpn73oFrovhIIox0TOw)

![image](https://github.com/user-attachments/assets/5a08bf0e-36e6-4603-9567-f84ab88bef10)
The error occured due to no dokploy image available in docker. I added a try/catch with automatic latest release pull and retry in case getting current digest returns an error. Could you try now please?

szwabodev commented:
> @szwabodev Hi, I'm getting this error
> 
> ![Screenshot 2024-12-20 at 11 33 01 PM](https://private-user-images.githubusercontent.com/47042324/397906972-719ed03f-2538-4dff-8d04-3caaf14a0b45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDY5NzItNzE5ZWQwM2YtMjUzOC00ZGZmLThkMDQtM2NhYWYxNGEwYjQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NDVlZjY5M2NhNWEwY2NjZTE0YTJlZWQyNzQ2NmI3NDIyYjhlMDU4MTViOTUxYTJiMDU3MzY3NmM2MGJiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aAEuCSxaWm8aVVhAQyICH8flPVzZABiCfW2ltQRua9U) ![Screenshot 2024-12-20 at 11 33 41 PM](https://private-user-images.githubusercontent.com/47042324/397907000-e766da6c-00ac-4af0-a2a2-8961f86781e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDcwMDAtZTc2NmRhNmMtMDBhYy00YWYwLWEyYTItODk2MWY4Njc4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZmYzZDdiZWU5YTExODdmM2RiZmY0YjI3MTJiMTRlYjFjNGQyMDAwMzQ5ZDgxNWEwM2MyNGUzN2Y5NzY2NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UXJ6SaoKLqPDWX58oiYY_yBP8VaT3UTUTxxx9OBEndo)

I refactored the logic to get image SHA from dokploy service instead of docker image (which was not pulled in your case causing this error). Could you please try now?

Siumauricio commented:
> > @szwabodev Hi, I'm getting this error
> > ![Screenshot 2024-12-20 at 11 33 01 PM](https://private-user-images.githubusercontent.com/47042324/397906972-719ed03f-2538-4dff-8d04-3caaf14a0b45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDY5NzItNzE5ZWQwM2YtMjUzOC00ZGZmLThkMDQtM2NhYWYxNGEwYjQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NDVlZjY5M2NhNWEwY2NjZTE0YTJlZWQyNzQ2NmI3NDIyYjhlMDU4MTViOTUxYTJiMDU3MzY3NmM2MGJiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aAEuCSxaWm8aVVhAQyICH8flPVzZABiCfW2ltQRua9U) ![Screenshot 2024-12-20 at 11 33 41 PM](https://private-user-images.githubusercontent.com/47042324/397907000-e766da6c-00ac-4af0-a2a2-8961f86781e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDcwMDAtZTc2NmRhNmMtMDBhYy00YWYwLWEyYTItODk2MWY4Njc4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZmYzZDdiZWU5YTExODdmM2RiZmY0YjI3MTJiMTRlYjFjNGQyMDAwMzQ5ZDgxNWEwM2MyNGUzN2Y5NzY2NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UXJ6SaoKLqPDWX58oiYY_yBP8VaT3UTUTxxx9OBEndo)
> 
> I refactored the logic to get image SHA from dokploy service instead of docker image (which was not pulled in your case causing this error). Could you please try now?
> 
> The error might still occur when dokploy service doesn't exist, but I guess we can assume that this service always exist when dokploy is running, right?
Now it give me this error
![Screenshot 2024-12-21 at 3 23 57 AM](https://github.com/user-attachments/assets/fccaeec4-4dea-4ebf-b067-c5b5a38d7675)
![Screenshot 2024-12-21 at 3 24 14 AM](https://github.com/user-attachments/assets/1f2f7b56-97e0-40bb-ad3e-2e6b10e3a64b)

szwabodev commented:
> > > @szwabodev Hi, I'm getting this error
> > > ![Screenshot 2024-12-20 at 11 33 01 PM](https://private-user-images.githubusercontent.com/47042324/397906972-719ed03f-2538-4dff-8d04-3caaf14a0b45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDY5NzItNzE5ZWQwM2YtMjUzOC00ZGZmLThkMDQtM2NhYWYxNGEwYjQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NDVlZjY5M2NhNWEwY2NjZTE0YTJlZWQyNzQ2NmI3NDIyYjhlMDU4MTViOTUxYTJiMDU3MzY3NmM2MGJiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aAEuCSxaWm8aVVhAQyICH8flPVzZABiCfW2ltQRua9U) ![Screenshot 2024-12-20 at 11 33 41 PM](https://private-user-images.githubusercontent.com/47042324/397907000-e766da6c-00ac-4af0-a2a2-8961f86781e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzIxOTAsIm5iZiI6MTczNDc3MTg5MCwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MDcwMDAtZTc2NmRhNmMtMDBhYy00YWYwLWEyYTItODk2MWY4Njc4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MDQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZmYzZDdiZWU5YTExODdmM2RiZmY0YjI3MTJiMTRlYjFjNGQyMDAwMzQ5ZDgxNWEwM2MyNGUzN2Y5NzY2NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UXJ6SaoKLqPDWX58oiYY_yBP8VaT3UTUTxxx9OBEndo)
> > 
> > 
> > I refactored the logic to get image SHA from dokploy service instead of docker image (which was not pulled in your case causing this error). Could you please try now?
> > The error might still occur when dokploy service doesn't exist, but I guess we can assume that this service always exist when dokploy is running, right?
> > Now it give me this error
> > ![Screenshot 2024-12-21 at 3 23 57 AM](https://private-user-images.githubusercontent.com/47042324/397917938-fccaeec4-4dea-4ebf-b067-c5b5a38d7675.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzM0OTUsIm5iZiI6MTczNDc3MzE5NSwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MTc5MzgtZmNjYWVlYzQtNGRlYS00ZWJmLWIwNjctYzViNWEzOGQ3Njc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MjYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMzliNDEzNGI0YmM5OGNkOGIwNGNlMWM1MDVhNjgzNmRhZTdkMTdiNDM5MjRiMDJkM2U0MjEzZDdlZWFlNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.H8_AFPNAP7q_eKqEFWJ4dxVVFHZLu4OKRWNwDMwocp4)
> > ![Screenshot 2024-12-21 at 3 24 14 AM](https://private-user-images.githubusercontent.com/47042324/397917950-1f2f7b56-97e0-40bb-ad3e-2e6b10e3a64b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3NzM0OTUsIm5iZiI6MTczNDc3MzE5NSwicGF0aCI6Ii80NzA0MjMyNC8zOTc5MTc5NTAtMWYyZjdiNTYtOTdlMC00MGJiLWFkM2UtMmU2YjEwZTNhNjRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIxVDA5MjYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3OTIwNTYxOTliY2M2ZTllM2NkNzBkYzRhN2JkMjQ0OTAyY2Q4Yjg5MTYyMjMzZTVlYjcwZTAwYzFlZjQ3N2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x9-BsGKISNlf9R48tZ23FKcN2FCf4BAmtx2SNEHeJGU)

Yeah that's what I asked if we can assume that dokploy service always exists. In base canary version there's similar error when trying to update the server due to missing dokploy service as well. I don't know if there is any possibility that dokploy service doesn't exist in self-hosted mode, except when running it locally which by default doesn't use dokploy service and you need to start it manually like I did for testing this.

szwabodev commented:
ok I added try/catch with default update data fallback, also added a comment with a link to manual installation docs for information how to start service locally to fully test checking updates functionality.

Siumauricio commented on apps/dokploy/components/layouts/navbar.tsx:
```diff
@@ -15,8 +15,13 @@ import { useRouter } from "next/router";
+import { useEffect, useRef, useState } from "react";
+import { UpdateWebServer } from "../dashboard/settings/web-server/update-webserver";
+
+const AUTO_CHECK_UPDATES_INTERVAL_MINUTES = 15;
```
```suggestion
const AUTO_CHECK_UPDATES_INTERVAL_MINUTES = 5;
```
